### ISSUE TITLE
False-positive between `ERA001` and `RUF100`

### ISSUE THREAD

**Issue by flbraun** [astral-sh/ruff#9551]
ruff 0.1.13

Take this piece of code:
```python
response = '...'  # imagine requests.get(...) here
# response.json() example:
# {
#   'foo': 1,
#   'bar': 2,
#   'baz': 3,
# }
bar = response.json()['bar']
```

`ruff check --select ERA001 --select RUF100` rightfully complains about commented-out code:
```
test.py:3:1: ERA001 Found commented-out code
test.py:4:1: ERA001 Found commented-out code
test.py:5:1: ERA001 Found commented-out code
test.py:6:1: ERA001 Found commented-out code
test.py:7:1: ERA001 Found commented-out code
Found 5 errors.
```

Then add noqa directives for ERA001, because we actually want to keep the comment:
```python
response = '...'  # imagine requests.get(...) here
# response.json() example:
# {                 # noqa: ERA001
#   'foo': 1,       # noqa: ERA001
#   'bar': 2,       # noqa: ERA001
#   'baz': 3,       # noqa: ERA001
# }                 # noqa: ERA001
bar = response.json()['bar']
```

`ruff check --select ERA001 --select RUF100` then complains about unused noqa directives, although they are definitely required to fully satisfy ERA001:
```
test.py:3:21: RUF100 [*] Unused `noqa` directive (unused: `ERA001`)
test.py:7:21: RUF100 [*] Unused `noqa` directive (unused: `ERA001`)
Found 2 errors.
```

Removing the supposedly unused noqa directives causes `ruff check --select ERA001 --select RUF100` to complain about ERA001 again:
```python
response = '...'  # imagine requests.get(...) here
# response.json() example:
# {
#   'foo': 1,       # noqa: ERA001
#   'bar': 2,       # noqa: ERA001
#   'baz': 3,       # noqa: ERA001
# }
bar = response.json()['bar']
```
```
test.py:3:21: RUF100 [*] Unused `noqa` directive (unused: `ERA001`)
test.py:7:21: RUF100 [*] Unused `noqa` directive (unused: `ERA001`)
Found 2 errors.
```

**Comment by charliermarsh** [comment]
Ah yeah this is tricky. The presence of the `# noqa` comments is causing Ruff _not_ to consider those lines as commented-out code.

**Comment by charliermarsh** [comment]
Thanks for the clear write-up.

**Comment by dylwil3** [comment]
This appears to work now: [Playground link](https://play.ruff.rs/d2082342-0c92-4df7-b727-e24bf84b9ae9)
